src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt

```diff
@@ -1843,6 +1844,14 @@ namespace NServiceBus.Features
     {
         protected internal override void Setup(NServiceBus.Features.FeatureConfigurationContext context) { }
     }
+    public class RoutingComponent
```

Comment:
I like this approach. I would lean to not postfix the components though, but that is nitpicking. In the future, we need to expose somehow the core infrastructure without needing to access DI. So for example right now if someone registers a behavior which needs to get access to routing or in the future the dispatcher the behavior registration would need to lean on DI. Assuming we had an interface on the feature configuration context like `ICoreComponents` which contains the routing, transport... we could instead of passing `IBuilder` into the lambda just pass in `ICoreComponents`. Of course, we could also say that the whenever you do something advanced like accessing the dispatcher we force users to write a feature. I would naturally lean towards not enforcing that though, but that is just my opinion.